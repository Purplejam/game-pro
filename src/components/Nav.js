import {motion, AnimateSharedLayout} from 'framer-motion';
import styled from 'styled-components';
import logo from '../img/logo.svg';
import {gamesSearchThunk, clearDataAction} from '../actions/gamesAction.js';
import {useState} from 'react';
import {useDispatch} from 'react-redux';

const StyledNav = styled(motion.div)`
	padding: 3rem 5rem;
 text-align: center;
 input {
   width: 30%;
   font-size: 1.5rem;
   padding: 0.5rem;
   border: none;
   margin-top: 1rem;
   box-shadow: 0px 0px 30px rgba(0, 0, 0, 0.2);
 }
 button {
   font-size: 1.5rem;
   border: none;
   padding: 0.5rem 2rem;
   cursor: pointer;
   background: #ff7676;
   color: white;
 }
`

const Logo = styled(motion.div)`
	display: flex;
	align-items: center;
	justify-content: center;
 padding: 1rem;
 cursor: pointer;
 img {
   height: 2rem;
   width: 2rem;
 }
`

const Search = styled(motion.form)`
	
`

export default function Nav() {
	const [inputValue, setInput] = useState('');
	const dispatch = useDispatch();

	const formSubmitHandler = async (e) => {
		e.preventDefault();
		await dispatch(gamesSearchThunk(inputValue));
		setInput('');
	}

	const clearInputHandler = () => {
		dispatch(clearDataAction());
		setInput('');
	}

	return(
		<StyledNav>
			<Logo onClick={clearInputHandler}>
				<img src={logo} alt="logo"/>
				<h1>Ignite</h1>
			</Logo>
			<Search onSubmit={formSubmitHandler}>
				<input value={inputValue} type="text" onChange={(e) => setInput(e.target.value)}/>
				<button type="submit">Search</button>
			</Search>
		</StyledNav>
		)
}