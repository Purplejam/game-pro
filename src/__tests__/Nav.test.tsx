import {render, screen, fireEvent} from '@testing-library/react';
import Nav from '../components/Nav';
import {store} from '../store/store';
import { Provider } from 'react-redux';
import React from 'react';
import { BrowserRouter} from "react-router-dom";
import '@testing-library/jest-dom';

 
describe('<Nav/>', () => {
	it('renders properly the game component', () => {
		const {container} = render(
			<BrowserRouter>
				<React.StrictMode>
					<Provider store={store}>
						 <Nav/>
					</Provider>
				</React.StrictMode>
			</BrowserRouter>
			)

		const logo = screen.getByAltText('logo');
		expect(logo).toBeInTheDocument();
		const input = screen.getByTestId<HTMLInputElement>('search-input');

		fireEvent.change(input, {target: {value: 'warcraft 3'}});
  expect(input.value).toBe('warcraft 3');

  //check if input is clear
  fireEvent.click(logo);
  expect(input.value).toBe('');

  //check if input has changed
  fireEvent.change(input, {target: {value: 'warcraft 12'}});
  expect(input.value).toBe('warcraft 12');
		expect(container).toMatchSnapshot();
	})
})

