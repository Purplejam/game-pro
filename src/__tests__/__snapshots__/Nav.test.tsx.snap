// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> renders properly the game component 1`] = `
<div>
  <div
    class="sc-bczRLJ dwLQWB"
  >
    <div
      class="sc-gsnTZi fhqrCy"
      data-testid="search-logo"
    >
      <img
        alt="logo"
        src="logo.svg"
      />
      <h1>
        Game PRO
      </h1>
    </div>
    <form
      class="sc-dkzDqf jlXTiK"
    >
      <input
        data-testid="search-input"
        type="text"
        value="warcraft 12"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
